### PR TITLE
Changed EmailInput to accept more generic inputs.

### LINKED ISSUE #1098: Accept a more generic input format in EmailInput
## issue

As per the discussion with @karthiknmenon and @Thejus-Paul 

The EmailInput component should accept input strings like `Mohit Srivastava <mohit.srivastava@bigbinary.com>` (Gmail pattern), when pasting from the clipboard and should be able to pluck out the email address.

### PR DESCRIPTION
**Pull request by mohit-s96**
closes #1098

### CODE REVIEW

**Comment by mohit-s96** [comment]
@Thejus-Paul _a. Please review.

**Review by Thejus-Paul — COMMENTED** [review]
LGTM.
@karthiknmenon _a Please review.

File: tests/formik/EmailInput.test.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ describe("formik/Input", () => {
     const onSubmit = jest.fn();
     render(<TestEmailInputForm onSubmit={onSubmit} />);
     const emailInput = screen.getByRole("combobox");
-    userEvent.paste(emailInput, "john.doe@email.com  sam.doeemail.com");
+    userEvent.paste(emailInput, "sam.doeemail.com");
```
Comment (ajmaln) [inline tests/formik/EmailInput.test.js:90]: @mohit-s96 is this change required?

Comment (mohit-s96) [reply tests/formik/EmailInput.test.js:90]: Hi,
The new format requires us to accept input like `John Doe <john@email.com>`, pick out `john@email.com` and discard the rest i.e. `John Doe <>`. Thus this input is considered valid (only if pasted through the clipboard). 

The issue here is that this test asserts that `john.doe@email.com  sam.doeemail.com` should be invalid when pasted from the clipboard, which wouldn't pass because there is at least one valid email present in the pasted input.

Another way of making it pass is using the `userEvent.type` method instead of `userEvent.paste`.

**Review by karthiknmenon — APPROVED** [review]
@amaldinesh7 _a Please review and merge.
